engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +60,27 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            this.containerId = containerId;
```

Comment:
containerId is _already stored_ in this class in `init` field - as soon as it is there. Storing it again is excessive. Where you need it, initi field shoulf be used.